### ISSUE TITLE
False positive `consider-using-dict-comprehension` when creating a dict using a list of tuple where key AND value vary depending on the same condition

### ISSUE THREAD

**Issue by tushar-deepsource** [PyCQA/pylint#5588]
### Bug description

The following code snippet raises R1717:

```python
dict([(k, v) if v > 0 else (k, 0) for k, v in stuff.items()])
```

Both of these contain an `IfExp`, which I don't think can be converted into a dict comprehension. Please correct me if I'm wrong.

### Command used

```shell
pylint mytest.py
```

### Pylint output

```shell
************* Module mytest
mytest.py:1:0: R1717: Consider using a dictionary comprehension (consider-using-dict-comprehension)
```

### Expected behavior

R1717 to not show up

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6-dev0
Python 3.10.0 (default, Oct 13 2021, 06:45:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```

**Comment by tushar-deepsource** [comment]
If this is actually a false positive, I'd be happy to submit a patch.

**Comment by Pierre-Sassoulas** [comment]
I think you can do : `{k: v if v > 0 else 0 for k, v in stuff.items()}`

**Comment by tushar-deepsource** [comment]
@Pierre-Sassoulas I don't think it generalizes, what if the key varies

`dict([(k, v) if v > 0 else (k * 2, 0) for k, v in stuff.items()])`

**Comment by Pierre-Sassoulas** [comment]
If there is the same condition used on both the key and the value, then it make sense to use tuple. (Because there's duplication with the dict comprehension : `{k if v > 0 else k * 2: v if v > 0 else 0 for k, v in stuff.items()}`).

**Comment by tushar-deepsource** [comment]
Now that's a very specific condition :P
I'll submit a patch soon

**Comment by Pierre-Sassoulas** [comment]
Haha, you called that an edge case, I think this was very descriptive :smile:  Thanks for handling this one !
